test/Core.js

```diff
@@ -19,38 +19,205 @@
+
+        it('has latestStateRootBlockHeight', async () => {
+            let actualBlockHeight = await core.getLatestStateRootBlockHeight.call();
+            assert.equal(actualBlockHeight.eq(blockHeight), true);
+        });
+
+    });
+
+
+    describe('commitStateRoot', async () => {
+        // Before All
+        before(async () => {
+            contractsData = await coreUtils.deployCore(artifacts, accounts);
+            core = contractsData.core;
+            worker = contractsData.worker;
+        });
+
+        it('should be able to commit state root and getStateRoot for give block height', async () => {
+            let stateRoot = '0x4567897545535535365000000000000000000000000000000000000000000000'
+                , blockHeight = 1
+            ;
+            let response = await core.commitStateRoot(blockHeight, stateRoot, {from: worker});
+
+            let formattedDecodedEvents = web3EventsDecoder.perform(response.receipt, core.address, core.abi);
+            let event = formattedDecodedEvents['StateRootCommitted'];
+
+            await coreUtils.checkStateRootCommittedEvent(event, blockHeight, stateRoot);
+            assert.equal(await core.getStateRoot(blockHeight), stateRoot);
+        });
+
+        it('should not be able to commit state root of block height if non worker commits root', async () => {
+            await utils.expectThrow(core.commitStateRoot(1, '0x4567897545535535365', {from: accounts[0]}));
+        });
+
+        it('should not be able to commit state root of block height which is already commited', async () => {
+            await utils.expectThrow(core.commitStateRoot(1, '0x4567897545535535365', {from: worker}));
+        });
+
+        it('should not be able to commit state root of block height less than latest block height', async () => {
+            await core.commitStateRoot(4, '0x45675567897545535535365', {from: worker});
+            await utils.expectThrow(core.commitStateRoot(3, '0x4567897545535535365', {from: worker}));
+        });
+
+        it('should not be able to commit state root of when state root is 0x', async () => {
+            await utils.expectThrow(core.commitStateRoot(5, '0x', {from: worker}));
+        });
+
+    });
+
+    describe('proveOpenST', async () => {
+        let blockHeight = 5
+            , parentNodes = ethUtil.rlp.decode(proof.account.rlpParentNodes)
+            , accountNode = parentNodes[parentNodes.length - 1]
+            , accountValue = ethUtil.rlp.decode(accountNode[1])
+            , storageRoot = '0x' + accountValue[2].toString('hex')
+        ;
+
+        before(async () => {
+
+            contractsData = await coreUtils.deployCore(artifacts, accounts);
+            core = contractsData.core;
+            worker = contractsData.worker;
+
+            await core.commitStateRoot(blockHeight, proof.account.stateRoot, {from: worker});
+        });
+
+        it('should not be able to verify proof for account if block height is 0', async () => {
```

Comment:
so this require will never fail.

This is not true. At time of initializing latestStateRootBlockHeight value is initialized as 0.
So if someone try commit a state root for 0, it will always fail as it need to be greater than 0.

If we want to allow commit a state root for 0, we should do at the time of deployment of contract and pass genesis block state root. However I am not sure, state root of block height 0 will ever be useful. Your thoughts?